### ISSUE TITLE
Last (automatic) snap refresh of rocketchat-server breaks service.

### ISSUE THREAD

**Issue by stst** [RocketChat/Rocket.Chat#19644]
### Description:

Automatic `snap refresh` installed latest snap `3.8.1` (Rev `1444`) and the Rocket.Chat instance was not accessible anymore.

Error in syslog:
```
Nov 21 14:27:21 chat rocketchat-server.rocketchat-server[8875]: /snap/rocketchat-server/1444/programs/server/npm/node_modules/meteor/promise/node_modules/meteor-promise/promise_server.js:218
Nov 21 14:27:21 chat rocketchat-server.rocketchat-server[8875]:       throw error;
Nov 21 14:27:21 chat rocketchat-server.rocketchat-server[8875]:       ^
Nov 21 14:27:21 chat rocketchat-server.rocketchat-server[8875]: TypeError: Cannot read property 'split' of null
Nov 21 14:27:21 chat rocketchat-server.rocketchat-server[8875]:     at app/lib/server/lib/validateEmailDomain.js:17:31
Nov 21 14:27:21 chat rocketchat-server.rocketchat-server[8875]:     at app/settings/lib/settings.ts:80:37
Nov 21 14:27:21 chat rocketchat-server.rocketchat-server[8875]:     at Array.forEach (<anonymous>)
Nov 21 14:27:21 chat rocketchat-server.rocketchat-server[8875]:     at app/settings/lib/settings.ts:80:15
Nov 21 14:27:21 chat rocketchat-server.rocketchat-server[8875]:     at Array.forEach (<anonymous>)
Nov 21 14:27:21 chat rocketchat-server.rocketchat-server[8875]:     at Settings.load (app/settings/lib/settings.ts:77:14)
Nov 21 14:27:21 chat rocketchat-server.rocketchat-server[8875]:     at Settings.storeSettingValue (app/settings/server/functions/settings.ts:342:8)
Nov 21 14:27:21 chat rocketchat-server.rocketchat-server[8875]:     at app/settings/server/functions/settings.ts:365:9
Nov 21 14:27:21 chat rocketchat-server.rocketchat-server[8875]:     at Array.forEach (<anonymous>)
Nov 21 14:27:21 chat rocketchat-server.rocketchat-server[8875]:     at Settings.init (app/settings/server/functions/settings.ts:364:32)
Nov 21 14:27:21 chat rocketchat-server.rocketchat-server[8875]:     at module (app/lib/server/startup/settings.js:2908:10)
Nov 21 14:27:21 chat rocketchat-server.rocketchat-server[8875]:     at fileEvaluate (packages/modules-runtime.js:336:7)
Nov 21 14:27:21 chat rocketchat-server.rocketchat-server[8875]:     at Module.require (packages/modules-runtime.js:238:14)
Nov 21 14:27:21 chat rocketchat-server.rocketchat-server[8875]:     at Module.moduleLink [as link] (/snap/rocketchat-server/1444/programs/server/npm/node_modules/meteor/modules/node_modules/reify/lib/runtime/index.js:52:22)
Nov 21 14:27:21 chat rocketchat-server.rocketchat-server[8875]:     at module (app/lib/server/index.js:1:153)
Nov 21 14:27:21 chat rocketchat-server.rocketchat-server[8875]:     at fileEvaluate (packages/modules-runtime.js:336:7)
```

unfortunately a `snap revert rocketchat-server` did not bring up a running instance, only a restore from last-evenings backup was working. 

### Steps to reproduce:

1. `snap refresh rocketchat-server` introduces the error above

### Expected behavior:

A snap refresh should not break the installation.

### Actual behavior:

rocket.chat instance does not start

### Server Setup Information:

- Version of Rocket.Chat Server: 3.8.1 (not working) - version before: 3.6.2 (also not working after snap refresh)
- Operating System: Ubuntu 18.04.5 LTS
- Deployment Method: snap
- Number of Running Instances: 1
- DB Replicaset Oplog: 
- NodeJS Version: snap-included
- MongoDB Version: snap-included

### Client Setup Information

No client is able to access, service is not running.

**Comment by stst** [comment]
Setting `snapctl set snap-refreshing=false` made 3.6.2 work again.

**Comment by betterphp** [comment]
We're having this issue too. It seems to be something to do with domains allow/deny lists for email addresses. We never used that feature so not sure if it's a new setting.

The error comes from this line https://github.com/RocketChat/Rocket.Chat/blob/cc28dbd9023e9e7deb152ab218b5da086e20c702/app/lib/server/lib/validateEmailDomain.js#L14

I came up with a bit of a hack to get it working without rolling back.

Copy the app bundle to a new folder
```
mkdir ~/rocketchat_server_app
cp /snap/rocketchat-server/1444/programs/server/app/* ~/rocketchat_server_app
```

Mount the new files on top of the ones in the snap package
```
mount --bind -o nodev,ro ~/rocketchat_server_app /snap/rocketchat-server/1444/programs/server/app/
```
This will make the snap use those files instead of the bundled ones. To revert just unmount that path.

Now we can edit the offending line in `/root/rocketchat_server_app/app.js:21402`

Make the two highlighted edits
![image](https://user-images.githubusercontent.com/788007/99950978-452e0700-2d75-11eb-983d-bb5bf9547330.png)

Restart the image
```
snap restart rocketchat-server
```

After a few seconds it should be back up and running.

Note that this will disable the email domain allow and deny lists, if that's a problem for you then don't do this.

**Comment by kevincolten** [comment]
I was able to mitigate it by setting some environment variables.

```
This is how you do it: https://docs.rocket.chat/installation/snaps#how-do-i-change-other-environmental-variables-in-my-snap

And these are the settings I need you to overwrite to try to avoid that split problem:

OVERWRITE_SETTING_Accounts_AllowedDomainsList=your-domain.com,gmail.com
OVERWRITE_SETTING_Accounts_BlockedDomainsList=somerandomdomain.com,anyotherdomain.com
```

**Comment by geekgonecrazy** [comment]
@betterphp thanks for sharing your work around.  I’ve hidden it though this will lead to issues down the road for anyone that follows it. 

If you are going to workaround it please follow @kevincolten suggestion.

Also looks like this did have a fix land in 3.9.0 which will land in snaps hopefully next week since it was a small release

**Comment by sampaiodiego** [comment]
The fix will be released on 3.8.2 as well. @geekgonecrazy I guess a snap release can be cut right away, wdyt?

**Comment by geekgonecrazy** [comment]
Yup for sure will release 3.8.2 to snap asap.  It won't be delayed any
